cirq/study/trial_result_test.py

```diff
@@ -21,42 +22,71 @@
 
 
 def test_repr():
-    v = cirq.TrialResult(
-        params=cirq.ParamResolver({'a': 2}),
-        repetitions=2,
-        measurements={'m': np.array([[1, 2]])})
+    v = cirq.TrialResult(params=cirq.ParamResolver({'a': 2}),
+                         measurements={'xy': np.array([[1, 0], [0, 1]])})
 
     assert repr(v) == ("cirq.TrialResult(params=cirq.ParamResolver({'a': 2}), "
-                       "repetitions=2, measurements={'m': array([[1, 2]])})")
+                       "repetitions=2, measurements={'xy': array([[1, 0],\n"
+                       "       [0, 1]])})")
 
 
 def test_str():
-    result = cirq.TrialResult(
-        params=cirq.ParamResolver({}),
-        repetitions=5,
-        measurements={
-            'ab': np.array([[0, 1],
-                            [0, 1],
-                            [0, 1],
-                            [1, 0],
-                            [0, 1]]),
-            'c': np.array([[0], [0], [1], [0], [1]])
-        })
+    result = cirq.TrialResult(params=cirq.ParamResolver({}),
+                              measurements={
+                                  'ab':
+                                  np.array([[0, 1], [0, 1], [0, 1], [1, 0],
+                                            [0, 1]]),
+                                  'c':
+                                  np.array([[0], [0], [1], [0], [1]])
+                              })
     assert str(result) == 'ab=00010, 11101\nc=00101'
 
 
+def test_df():
+    result = cirq.TrialResult(params=cirq.ParamResolver({}),
+                              measurements={
+                                  'ab':
+                                  np.array(
+                                      [[0, 1], [0, 1], [0, 1], [1, 0], [0, 1]],
+                                      dtype=np.bool),
+                                  'c':
+                                  np.array([[0], [0], [1], [0], [1]],
+                                           dtype=np.bool)
+                              })
+    remove_end_measurements = pd.DataFrame(data={
+        'ab': [
+            pd.Series([False, True]),
+            pd.Series([False, True]),
+            pd.Series([True, False])
+        ],
+        'c': [pd.Series([False]),
+              pd.Series([True]),
+              pd.Series([False])]
+    },
+                                           index=[1, 2, 3])
+
+    pd.testing.assert_frame_equal(result.data.iloc[1:-1],
+                                  remove_end_measurements)
+
+    ab = result.data['ab']
+    c = result.data['c']
+    # Count number of ab repitiions with measurements [False, True]
```

Comment:
I think this use case might be common enough to provide a util for. WDYT?

Reply:
That's what the histogram methods are supposed to do, though I expect they'll be superceded.